src/main/java/lotto/domain/LottoResultStatistics.java

```diff
@@ -0,0 +1,49 @@
+package lotto.domain;
+
+import lotto.domain.lottos.LottoTickets;
+import lotto.domain.lottos.rank.LottoRank;
+import lotto.domain.lottos.winnerlotto.LottoWinner;
+import lotto.domain.money.Money;
+
+import java.util.Arrays;
+import java.util.EnumMap;
+import java.util.Map;
+
+public class LottoResultStatistics {
+    private static final int EARNING_RATE = 100;
+    private static final int DEFAULT_COUNT = 0;
+
+    private final Map<LottoRank, Integer> lottoResult;
+
+    private LottoResultStatistics(final Map<LottoRank, Integer> lottoResult) {
+        this.lottoResult = lottoResult;
+    }
+
+    public static LottoResultStatistics calculateResultStatistics(
+            final LottoTickets lottoTickets, final LottoWinner lottoWinner) {
+        Map<LottoRank, Integer> lottoResult = setLottoResult();
+        lottoTickets.putLottoResult(lottoResult, lottoWinner);
+        return new LottoResultStatistics(lottoResult);
+    }
+
+    private static Map<LottoRank, Integer> setLottoResult() {
+        Map<LottoRank, Integer> lottoResult = new EnumMap<>(LottoRank.class);
+        Arrays.stream(LottoRank.values())
+                .forEach(value -> lottoResult.put(value, DEFAULT_COUNT));
```

Comment:
`Collection.foreach`와 `Stream.foreach`에 관해 한번 찾아보면 좋을 것 같아요! 

```suggestion
        Arrays.asList(LottoRank.values())
                .forEach(value -> lottoResult.put(value, DEFAULT_COUNT));
```

Reply:
해당내용에 대해서 학습해보니
여기서는 Stream의 생성이 낭비되고 있음을 알게됐어요.
감사합니다 데이브 🙇‍♂️ 

791806c

추가로 Collection.forEach는 쓰레드 세이프하다는걸 알았고
Stream.forEach는 병렬 프로그래밍에 특화된 반복을 한다는걸 알았어요!

https://github.com/woowacourse/java-lotto/pull/265#issuecomment-782615289